projects/plugins/boost/app/assets/src/js/pages/purchase-success/PurchaseSuccess.svelte

```diff
@@ -15,9 +15,17 @@
 	// svelte-ignore unused-export-let - Ignored values supplied by svelte-navigator.
 	export let location, navigate;
 
-	onMount( async () => {
+	onMount( () => {
+		// Enable cloud css module on upgrade.
 		$modulesState.cloud_css.active = true;
-		await regenerateCriticalCss();
+
+		// Ones the module is enabled and synced, regenerate critical css.
+		const unsubscribe = modulesStatePending.subscribe( pending => {
+			if ( ! pending ) {
+				regenerateCriticalCss();
+				unsubscribe();
+			}
+		} );
```

Comment:
Wouldn't it make more sense to make the plugin back-end automatically request fresh Cloud CSS when that module is enabled? We have an action on the back end `jetpack_boost_module_status_updated` which seems like it would be good for this purpose.

Then we don't have to do complex awaiting on the front end.

Reply:
I have addressed this in 2831905